MODULE.bazel

```diff
@@ -10,14 +10,21 @@ bazel_dep(name = "bazel_skylib", version = "1.4.1")
 bazel_dep(name = "rules_proto", version = "6.0.0")
 bazel_dep(name = "platforms", version = "0.0.8")
 
-protoc = use_extension("//protoc:extensions.bzl", "protoc")
+protoc = use_extension(
+    "//protoc:extensions.bzl",
+    "protoc",
+    dev_dependency = True,
```

Comment:
This is one way to solve it, but as you see it requires the user to do more setup work.

Instead, we should resolve the error by ensuring that the root module gets their toolchain registered first, so it takes precedence over a transitive one.

Do you have a repro of the error?

Reply:
That's not really possible since dependencies get priority. To repro, create a repository with the following `MODULE.bazel`:

```
module(name="test")

bazel_dep(name = "toolchains_protoc", version = "0.2.4")

protoc = use_extension("@toolchains_protoc//protoc:extensions.bzl", "protoc")
protoc.toolchain(version = "v27.0-rc3")
use_repo(protoc, "toolchains_protoc_hub")
```

and try to compile any proto file using a language toolchain.

Reply:
No, the bzlmod extension implementation has a global view of all registrations, so it's just a logic bug if dependencies get priority. The root module should always win.

Reply:
> No, the bzlmod extension implementation has a global view of all registrations, so it's just a logic bug if dependencies get priority. The root module should always win.

You're right, I was misunderstood something about this bug. I've found a workaround below though.